src/test.rs

```diff
@@ -417,4 +417,28 @@ mod tests {
 
         p!(j.join());
     }
+
+    #[test]
+    fn badssl_cipher_suites_no_rsa() {
+        let builder = p!(TlsConnector::builder().supported_cipher_suites(
+            // Oddly, on Windows, allowing RSA key exchange, but not RSA signature algorithms still
```

Comment:
This is the kind of edge case I'm worried about in exposing this level of control over cipher suites.

Reply:
FWIW, RSA is the only one that's valid in two different parts of the ciphersuite specification, so it's the only one this can happen to. Did some searching and it seems like it might be an expected quirk of Schannel that `RSA_KEYX` implies `RSA_SIGN`, although all the discussions that I found were on tangential topics ([1], [2]).

[1]: https://stackoverflow.com/questions/45211765/cryptoapi-rsa-schannel-provider-error
[2]: https://stackoverflow.com/questions/3568695/ms-csp-difference-between-at-signature-and-rsa-key-sign-and-also-at-keyexchang